core/src/impl/Kokkos_Profiling.hpp

```diff
@@ -59,7 +59,31 @@ namespace Kokkos {
 bool tune_internals() noexcept;
 
 namespace Tools {
-
+namespace Impl {
+template <class T>
+struct TuningResult;
+template <class Policy>
+struct TuningResult {
+  using type = Policy;
+};
+template <class input>
+struct translate {
+  using type = input;
+};
+template <>
+struct translate<Kokkos::Experimental::TuneOccupancy> {
+  using type = Kokkos::Experimental::DesiredOccupancy;
+};
+template <class... Properties>
+struct TuningResult<Kokkos::RangePolicy<Properties...>> {
+  // using type = Kokkos::RangePolicy<Properties...>;
+  using type = Kokkos::RangePolicy<typename translate<Properties>::type...>;
+};
+template <typename PolicyType, typename Functor>
+struct ToolResponse {
+  typename TuningResult<PolicyType>::type policy;
```

Comment:
The one element struct here is used so that when we need the tools subsystem to return more than one thing, we can extend it easily